vllm/v1/worker/gpu_model_runner.py

```diff
@@ -828,6 +843,12 @@ def _prepare_inputs(
                 causal=True,
             )
 
+            is_enc_dec = isinstance(kv_cache_group_spec.kv_cache_spec,
```

Comment:
don't we know this is an enc_dec model already?

Reply:
We do, but here we need to know if we're looking at a cross-attention layer or regular self-attention. So the check is about attention type, not the model type. We use `AttentionType.ENCODER_DECODER` so the attention type and model type share the same name in our code ... it's a little confusing sometimes.